consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/backward/sync/BackwardsSyncContext.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.merge.blockcreation.backward.sync;
+
+import org.hyperledger.besu.ethereum.BlockValidator;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.eth.manager.EthContext;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class BackwardsSyncContext {
+  private final ProtocolContext protocolContext;
+  private final ProtocolSchedule protocolSchedule;
+  private final BlockValidator blockValidator;
+  private final EthContext ethContext;
+  private final MetricsSystem metricsSystem;
+
+  private final Map<Long, BackwardChain> backwardChainMap = new ConcurrentHashMap<>();
+  AtomicReference<Block> currentPivot = new AtomicReference<>();
+
+  public BackwardsSyncContext(
```

Comment:
Ideally we would extend SyncState with a merge-specific syncState and just use the MergeContext for access to protocolContext and protocolSchedule.  We could add metrics and ethcontext into the merge-specific sync state class if we need.

That will allow us to look at just a single place when we need to check if we are syncing - either forward or backward, when we are executing blocks or updating fork choice

Reply:
On second thought, having backwards sync and forward sync be separate works well with the notion of appending new blocks to the end of the backward sync from `engineExecuteBlock`.  The only place we have to check both syncState and backwardsSyncContext is in `engineForkChoiceUpdated`.